presto-main/src/main/java/io/prestosql/sql/planner/QueryPlanner.java

```diff
@@ -162,29 +174,52 @@ public RelationPlan plan(QuerySpecification node)
 
         builder = window(builder, node);
 
-        List<Expression> outputs = analysis.getOutputExpressions(node);
-        builder = handleSubqueries(builder, node, outputs);
+        List<SelectExpression> selectExpressions = analysis.getSelectExpressions(node);
+        List<Expression> expressions = selectExpressions.stream()
+                .map(SelectExpression::getExpression)
+                .collect(toImmutableList());
+        List<Expression> outputs = expressions;
+        builder = handleSubqueries(builder, node, expressions);
+        List<Expression> outputExpressions = outputExpressions(selectExpressions);
 
         if (node.getOrderBy().isPresent()) {
             if (!analysis.isAggregation(node)) {
                 // ORDER BY requires both output and source fields to be visible if there are no aggregations
-                builder = project(builder, outputs, fromRelationPlan);
-                outputs = toSymbolReferences(computeOutputs(builder, outputs));
+                builder = project(builder, Iterables.concat(expressions, toSymbolReferences(fromRelationPlan.getFieldMappings())));
```

Comment:
Can you add some tests containing `xxx.*` in queries involving aggregations and/or order by?